Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride.hs

```diff
@@ -153,3 +176,108 @@ arrivedAtPickup rideId req = do
   pure Success
   where
     isValidRideStatus status = status == DRide.NEW
+
+otpRideCreate ::
+  ( HasCacheConfig r,
+    EsqDBFlow m r,
+    EsqDBReplicaFlow m r,
+    Redis.HedisFlow m r,
+    HasFlowEnv m r '["nwAddress" ::: BaseUrl],
+    HasFlowEnv m r '["selfUIUrl" ::: BaseUrl],
+    HasDriverPoolConfig r,
+    HasHttpClientOptions r c,
+    EncFlow m r,
+    HasShortDurationRetryCfg r c,
+    CoreMetrics m,
+    HasPrettyLogger m r
+  ) =>
+  DP.Person ->
+  OTPRideReq ->
+  m DriverRideRes
+otpRideCreate driver req = do
+  now <- getCurrentTime
+  bookingId <- runInReplica $ QBooking.findBookingBySpecialZoneOTP req.specialZoneOtpCode now >>= fromMaybeM (BookingNotFoundForSpecialZoneOtp req.specialZoneOtpCode)
+  booking <- runInReplica $ QBooking.findById bookingId >>= fromMaybeM (BookingDoesNotExist bookingId.getId)
+  transporter <-
+    QM.findById booking.providerId
+      >>= fromMaybeM (MerchantNotFound booking.providerId.getId)
+
+  driverInfo <- QDriverInformation.findById (cast driver.id) >>= fromMaybeM DriverInfoNotFound
+  when driverInfo.onRide $ throwError DriverOnRide
+  let otpCode = req.specialZoneOtpCode
```

Comment:
where are you verifying the otp provided by driver is correct?

Reply:
I am fetching the booking through top only.
so If otp will be incorrect booking will not come in findBooking query.
It will not even come to this line.